### PR TITLE
protobuf@3.1: remove

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Remove protobuf 3.1 as it is quite old.

This also removes supersonic as it is unmaintained, and there are no installs in the last 90 days.

This also bumps a revision for `ola` and migrates it to `protobuf@3.6`. This currently requires a patch, see https://github.com/OpenLightingProject/ola/issues/1192 from 2017. Protobuf 3.7+ support doesn't come with `ola` yet (see https://github.com/OpenLightingProject/ola/pull/1558) so we use 3.6 for now.

### CODE REVIEW

**Comment by taquitos** [comment]
Is there any reason besides age that this is removed? I ask because this breaks us. We are pinned to this release due to sharing protos between platforms we can't easily update our protos to be compatible with newer releases.
https://github.com/google/science-journal-ios

**Comment by fxcoudert** [comment]
The criteria for versioned formulas are quite strict: https://docs.brew.sh/Versions
If you think we made a mistake in assessing this, and the formula meets the criteria, we can revisit this decision.

Otherwise, it seems like the best thing for you would be to maintain your own specific version in a tap: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
Homebrew does not commit to maintaining older versions of software over time

**Comment by jonchang** [comment]
As FX said. Moreover, protobuf 3.1 was only included for use by another homebrew formula that couldn’t update. Now that that’s no longer the case, it got removed. See the last paragraph of the versions document.

**Comment by taquitos** [comment]
Thanks for the responses!

File: Formula/ola.rb [hunk 1]
```diff
@@ -3,45 +3,35 @@ class Ola < Formula
...
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
   depends_on "pkg-config" => :build
   depends_on "liblo"
   depends_on "libmicrohttpd"
   depends_on "libusb"
-  depends_on "numpy@1.16"
-  depends_on "protobuf@3.1"
-  depends_on "python@2" # protobuf@3.1 does not support Python 3
```
Comment (peternewman) [inline Formula/ola.rb]: @jonchang we still have some bugs with Python 3:
Issue: https://github.com/OpenLightingProject/ola/issues/1506
WIP PR: https://github.com/OpenLightingProject/ola/pull/1538

**Comment by peternewman** [comment]
> Remove protobuf 3.1 as it is quite old.
> 
> This also bumps a revision for `ola` and migrates it to `protobuf@3.6`. This currently requires a patch, see [OpenLightingProject/ola#1192](https://github.com/OpenLightingProject/ola/issues/1192) from 2017. Protobuf 3.7+ support doesn't come with `ola` yet (see [OpenLightingProject/ola#1558](https://github.com/OpenLightingProject/ola/pull/1558)) so we use 3.6 for now.

Ironically of course this has broken OLA's 0.10 based Mac CI! :smile:

File: Formula/ola.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/ola.rb]: Hopefully you will have a new release with better support soon, then :)

**Pull request closed without merging**